dripline/extensions/ethernet_thermo_fisher_service.py

```diff
@@ -0,0 +1,172 @@
+import re
+import socket
+import threading
+import time
+
+from dripline.core import Service, ThrowReply
+
+import logging
+logger = logging.getLogger(__name__)
+
+__all__ = []
+
+
+__all__.append('EthernetThermoFisherService')
+
+def int_to_hexstr(value):
+    return hex(value)[2:].zfill(2)
+
+def hexstr_to_bytes(value):
+    return bytes.fromhex(value)
+
+def bytes_to_hexstr(value):
+    return value.hex()
+
+def hexstr_to_int(value):
+    return int(value, 16)
+
+def bytes_to_ints(value):
+    return [byte for byte in value]
+
+class EthernetThermoFisherService(Service):
```

Comment:
I'm wondering if it would make sense to inherit from `dripline.EthernetSCPIService` and reimplement `_send_commands()` (we could think about restructuring `EthernetSCPIService._send_commands()` to use `_assemble_cmd()`, which would further simplify this class), and `_listen()`, plus adding the other necessary methods.

Reply:
If I revert back to use the `_reconnect` function I can inherit from EhternetSCPIService. I anyway have to overwrite the __init__ due to different arguments that are needed but then only _send_commands() would need modifications.